pkg/asset/installconfig/azure/validation.go

```diff
@@ -123,17 +123,18 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 
 	defaultDiskType := aztypes.DefaultDiskType
 	defaultInstanceType := ""
```

Comment:
naming nit: can we change this to something like `defaultMachinePlatformInstanceType` or `defaultMPInstanceType`? I think we should distinguish that this is a default set by the user rather than the default chosen by the installer in the `defaults` package.